lib/ddtrace/runtime/identity.rb

```diff
@@ -1,48 +0,0 @@
-require 'securerandom'
-require 'ddtrace/ext/runtime'
-require 'ddtrace/utils/forking'
-
-module Datadog
-  module Runtime
-    # For runtime identity
-    module Identity
-      extend Datadog::Utils::Forking
-
-      module_function
-
-      # Retrieves number of classes from runtime
-      def id
-        @id ||= SecureRandom.uuid
-
-        # Check if runtime has changed, e.g. forked.
-        after_fork! { @id = SecureRandom.uuid }
-
-        @id
-      end
```

Comment:
I'm somewhat concerned about this claim:

> These files that have been moved are not part of our public interface, thus will have no impact on users.

...because our current interfaces are not documented as "being public" and "not being public". So, as a advanced user... how should I know not to use this class?

Note that I don't think this state of matters should make us afraid of doing changes. But I think assuming that all these things are private and thus safe to move without that being clearly stated is being slightly _too_ optimistic.

Thus, I suggest the following:
* Let's document this as a possible breaking change in a very clear way in the PR description, and include that in the changelog once we release it
* Let's start documenting that these are private, to avoid any issues in the future. [Perhaps using a doc tag](https://rubydoc.info/gems/yard/file/docs/Tags.md#visibility)? Or adding a note on every file that should or should not be seen as private? Or having a general "API notes" in the documentation that states that you shouldn't reference anything from `Datadog::Core` and that it's expected to break even in minor patch releases? Or a combination of them?

Reply:
Documenting what's public in `ddtrace` is a huge gap we have today. Totally agree, also I agree with having it be in code (with a doc tag).

I'll update this PR description with a clear changelog language for the release as well.

Reply:
To be honest this one one of the things that worries me the most about the future of dd-trace.

That, and how it's too customizable -- users can inject behavior and bypass our code at so many different points, and while that isn't bad from a "if you want to have something really custom you can...", we don't clearly document which injection points are considered advanced tools that are unsupported, and so they become another really weird and hard to support public API, e.g. #1560.